src/Behat/Context/MediaContext.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+use Metadrop\Behat\Context\RawDrupalContext;
+use Metadrop\Behat\Context\UIContext;
+use Metadrop\Behat\Context\WaitingContext;
+use Behat\Gherkin\Node\TableNode;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+
+/**
+ * Steps related with media.
+ */
+class MediaContext extends RawDrupalContext {
+
+  protected $medias = [];
+
+  /**
+   * Mink context.
+   *
+   * @var Drupal\DrupalExtension\Context\MinkContext
+   */
+  protected $minkContext;
+
+  /**
+   * UI context.
+   *
+   * @var Metadrop\Behat\Context\UIContext;
+   */
+  protected $uiContext;
+
+  /**
+   * UI context.
+   *
+   * @var Metadrop\Behat\Context\WaitingContext;
+   */
+  protected $waitingContext;
+
+  /**
+   * Get the necessary contexts.
+   *
+   * @BeforeScenario
+   *
+   * @param BeforeScenarioScope $scope
+   *   Scope del scenario.
+   */
+  public function gatherContexts(BeforeScenarioScope $scope) {
+    $environment = $scope->getEnvironment();
+    $this->minkContext = $environment->getContext('Drupal\DrupalExtension\Context\MinkContext');
+    $this->uiContext = $environment->getContext(UIContext::class);
+    $this->waitingContext = $environment->getContext(WaitingContext::class);
+    $this->mediaContext = $environment->getContext(MediaContext::class);
+  }
+
+  /**
+   * Delete created medias.
+   *
+   * @AfterScenario
+   */
+  public function deleteMedias() {
+    foreach ($this->medias as $media) {
+      $this->getCore()->entityDelete('media', $media);
+    }
+  }
+
+  /**
+   * Select existing media.
+   *
+   * @Then I select media :arg1
+   */
+  public function iSelectMedia($widget, $xpath) {
+    if ($widget == 'media_library') {
+      $this->getSession()->getPage()->find('xpath', $xpath)->click();
+      $this->getSession()->getPage()->find('css', '.ui-dialog-buttonpane.ui-widget-content .media-library-select.form-submit')->click();
+    }
```

Comment:
The method should throw a [invalid argument exception](https://www.php.net/manual/en/class.invalidargumentexception.php) when the widget is not media_library, as it would do nothing